File: apps/web/src/features/profile/api/get-profile.ts
import { api } from '@/api/client'
import type { Profile } from '@/features/profile/profile'

export async function getProfile(): Promise<Profile> {
  const { data, error } = await api.GET('/api/v1/profile')
  if (error) throw new Error('Failed to load profile')
  return data
}

File: apps/api/internal/profile/application/get_profile.go
package application

import (
	"caudal-api/internal/profile/domain"
	"caudal-api/internal/profile/ports"
	"context"
)

type GetProfile struct{ repository ports.Repository }

func NewGetProfile(repository ports.Repository) GetProfile { return GetProfile{repository: repository} }
func (useCase GetProfile) Execute(ctx context.Context) (domain.Profile, error) {
	return useCase.repository.GetOrCreate(ctx)
}
